src/addon-manager.js

```diff
@@ -760,12 +760,14 @@ class AddonManager extends EventEmitter {
       this.adapters.delete(a.id);
     }
 
-    const plugin = this.getPlugin(packageName);
-    if (plugin) {
-      plugin.kill();
-    }
+    const cleanup = () => {
+      const plugin = this.getPlugin(packageName);
+      if (plugin) {
+        plugin.kill();
+      }
+    };
 
-    return Promise.all(unloadPromises);
+    return Promise.all(unloadPromises).then(() => cleanup(), () => cleanup());
```

Comment:
I still we think a delay here. The client side of the plugin waits for 1/2 second after sending the plugin unloaded message before closing the IPC connections (because the client doesn't know that the server received the message and closing will cause the message to disappear).